packages/EasyCore/tests/Bridge/Symfony/Valiator/Constraints/AbnValidatorTest.php

```diff
@@ -0,0 +1,296 @@
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+/**
+ * @covers \EonX\EasyCore\Bridge\Symfony\Validator\Constraints\AbnValidator
+ */
+final class AbnValidatorTest extends AbstractSymfonyTestCase
+{
+    /**
+     * @return string[][]
+     */
+    public function provideAbnWithInvalidCharacters(): array
+    {
+        return [
+            'Contains letter' => ['1234567890a'],
+            'Contains special symbols' => ['1234(678900'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideUnexpectedValues(): array
+    {
+        return [
+            'class without __toString' => [new stdClass(), 'Expected argument of type "string", "stdClass" given'],
+            'integer' => [53004085616, 'Expected argument of type "string", "integer" given'],
+        ];
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function provideValidAbn(): array
+    {
+        return [
+            'Valid Abn #1' => ['53004085616'],
+            'Valid Abn #2' => ['28043145470'],
+            'Valid Abn #3' => ['91724684688'],
+            'Valid Abn #4' => ['10000000000'],
+        ];
+    }
+
+    /**
+     * @return mixed[]
+     */
+    public function provideValidEmptyValues(): array
+    {
+        return [
+            'Empty string' => [''],
+            'Null value' => [null],
+        ];
+    }
+
+    /**
+     * @return string[][]
+     */
+    public function provideInvalidAbn(): array
+    {
+        return [
+            'Invalid Abn #1' => ['10043145470'],
+            'Invalid Abn #2' => ['53004085615'],
+            'Invalid Abn #3' => ['53004085615'],
+            'Invalid Abn #4' => ['10000000001'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideValidAbn
+     */
+    public function testValidateSucceedsWithValidAbn(string $abn): void
+    {
+        $validator = new AbnValidator();
+        $constraint = new Abn();
+        $context = $this->mockExecutionContextWithoutCalls();
+        $validator->initialize($context);
+
+        $validator->validate($abn, $constraint);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    public function testValidateSucceedsWithObjectToString(): void
+    {
+        $class = new class() {
+            public function __toString()
+            {
+                return '53004085616'; // Valid Abn
+            }
+        };
+        $validator = new AbnValidator();
+        $constraint = new Abn();
+        $context = $this->mockExecutionContextWithoutCalls();
+        $validator->initialize($context);
+
+        $validator->validate($class, $constraint);
+
+        $this->expectNotToPerformAssertions();
+    }
+
+    /**
+     * @dataProvider provideValidEmptyValues
+     *
+     * @param mixed $abn
+     */
+    public function testValidateSucceedsWithEmptyValue($abn): void
```

Comment:
This method completely repeats `testValidateSucceedsWithValidAbn`.

Reply:
Done.